## 1. 비즈니스 이해

### Background: 어떠한 문제가 존재하는지 상황에 관해 알려진 기사, 논문, 보고서 등을 인용하여 현상을 구체적으로 기술하시기 바랍니다.
- [국내 찾는 외국인 더 줄었다...1월 여행수지 적자 최대](http://www.ajunews.com/view/20180306083226894)
- 지난 1월(’18) 겨울 성수기를 맞아 해외여행을 다녀온 내국인이 늘었다. 반면 국내를 찾는 해외 관광객들의 발걸음은 뜸해지며 여행수지 적자가 역대 최대를 기록했다.
- 실제로 겨울방학을 맞아 해외 출국자 수는 286만 7,000명으로 역대 최대를 기록했다. 1년 전보다 22.4% 증가했다. 이에 반해 전체 입국자 수는 95만 6,000명으로 21.7% 감소했다. 특히 중국인 입국자 수는 중국의 사드(THAAD•고고도미사일방어체계) 배치 보복 완화에도 전년 대비 46.0% 감소한 30만 5,000명에 그쳤다.

-[제주관광, 외국인 52% 뚝… 11% 늘어난 내국인이 떠받쳤다] (http://news.donga.com/View?gid=83574500&date=20170329)
-이를 위해 전문가들은 중국 이외의 국가에서 외국인 관광객을 유치하고 국내 관광을 활성화해 중국인 관광객 급감 현상을 극복해야 한다고 강조한다.
-실제로 국내 관광이 크게 늘어날 경우 중국인 관광객의 공백을 상당 부분 메울 수 있는 것으로 기대된다. 한국수출입은행에 따르면 중국인 관광객이 60% 급감하는 최악의 시나리오를 가정했을 때 관광수입 손실액은 연간 5조5000억 원에 이를 것으로 추정된다.

-![추이](추이.jpg)
-![효과](효과.jpg)

### Business Objectives: 기업(혹은 본인)에 이 현상이 어떠한 이유로 분석해야 하는지 추상적(함축적)으로 기술하시기 바랍니다.
- 국내여행을 온 외국인 관광객들의 만족도, 지출 경비, 재방문 의사, 반응 등을 다양하게 파악하여 정부의 입장에서 외국인 관광객 유치에 도움이 되는 정보를 드리고, 외국인 대상 국내 관광 산업 활성화로 말미암은 국내 여행수지 흑자에 도움이 되고자 한다. 


### Business Success Criteria: 비즈니스 관점에서 분석결과의 성공기준(정량적) 또는 유용한 결과에 관한 기준(정성적)을 설명하십시오.
- 2011~2016년 외래관광객 실태조사 결과 중 국내여행을 온 외국인 관광객들의 총지출경비, 만족도, 재방문의사, 관광지별 반응 등의 지표로 회귀분석, 산점도 분석, 워드클라우드 분석, 엑셀 분석을 사용하여 유의미한 결과값이 나오도록 한다.

### Analysis Goals: 비즈니스 목표에 관한 의사결정 지원을 위해 실시하는 분석의 목표를 기술합니다. 
매년 약 12,000개의 Raw Data 데이터를 통해 회귀분석, 산점도 분석, 워드클라우드 분석, 엑셀 분석을 사용하여 유의미한 결과 값을 도출, 정부에 도움이 되는 정보를 제공한다. 


## 2. 데이터의 이해
	
- <외래관광객 실태조사> 7개년 원자료 파일을 통해 데이터를 수집함.
- [2016 외래관광객 실태조사](https://kto.visitkorea.or.kr/kor/notice/data/statis/tstatus/forstatus/board/view.kto?id=428253&rnum=2)
    - 데이터: 2011년 ~ 2016년 <외래관광객 실태조사> 원자료 (한국관광공사)
    - 설문: 외래관광객 대상, 매년 1회 실시(매년 약 12,000명)
- 데이터 전처리: 3년 내 재방문 의사, 만족도 변수 10개, 추천 의사, 외국인 관광객 리뷰 댓글, 총지출 경비


### 분석한 내용:
- 1) 회귀분석: 만족도 & 3년 이내 재방문 의사 / 만족도 & 추천의사(2013~2016)
- 2) 산점도: 재방문의사, 추천의사에 대한 위 회귀분석 자료를 토대로 분석(2016)
- 3) 워드클라우드 분석: 트립어드바이저(Tripadvisor) 사이트 리뷰 댓글을 취합하여 국내 인기 명소를 다녀온 외국인 여행객의 반응 분석
- 4) 엑셀분석: 총지출경비 자료로 평균지출경비, 최대지출경비 분석(2016)
- 5) 엑셀분석: 만족도, 재방문의사, 추천의사 응답의 각 문항 별 인원 수, 백분율 분석 (2013~2016)
- 6) 엑셀분석: 3년 내 한국 관광 재방문 의사 중 만족 의사 백분율 분석 (2011~2016)


In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import networkx as nx
import numpy as np
# import pydot
%matplotlib inline

# 폰트 문제 해결하기
import platform

path = "c:/Windows/Fonts/malgun.ttf"
from matplotlib import font_manager, rc
if platform.system() == 'Darwin':
    rc('font', family='AppleGothic')
elif platform.system() == 'Windows':
    font_name = font_manager.FontProperties(fname=path).get_name()
    rc('font', family=font_name)
else:
    print('Unknown system... sorry~~~~') 

In [2]:
data = pd.read_excel("dd.xlsx")
data.head()

X1  X2  X3   X4  X5  X6  X7  X8  X9  X10  Y1  Y2
0   3   5   4  5.0   3   3   4   3   3    5   3   3
1   4   4   3  4.0   4   5   5   3   3    5   4   4
2   5   5   5  5.0   5   5   5   5   5    5   5   5
3   4   4   4  5.0   4   3   4   3   4    3   4   4
4   3   4   5  5.0   5   4   4   4   4    4   5   5

In [3]:
data = data.dropna(axis=1) # axis =1 열을 기준으로 삭제
data.head()

X1  X2  X3  X5  X6  X7  X8  X9  X10  Y1  Y2
0   3   5   4   3   3   4   3   3    5   3   3
1   4   4   3   4   5   5   3   3    5   4   4
2   5   5   5   5   5   5   5   5    5   5   5
3   4   4   4   4   3   4   3   4    3   4   4
4   3   4   5   5   4   4   4   4    4   5   5

In [4]:
import statsmodels.api as sm
import statsmodels.formula.api as smf
import statsmodels.stats.api as sms

In [19]:
from sklearn import preprocessing
import pandas as pd

X=pd.DataFrame(data, columns=["X1","X2","X3","X5","X6","X7","X8","X9","X10"])
Y=pd.DataFrame(data, columns=["Y1"])


In [22]:
# 만족도와 재방문의사 회귀분석
linear_regression = sm.OLS(Y, X) # (종속변수, 독립변수)

# OLS 회귀모형 피팅
result_data = linear_regression.fit()

# 회귀분석 결과 요약
print(result_data.summary())

                            OLS Regression Results                            
Dep. Variable:                     Y1   R-squared:                       0.967
Model:                            OLS   Adj. R-squared:                  0.967
Method:                 Least Squares   F-statistic:                 3.914e+04
Date:                Sat, 09 Jun 2018   Prob (F-statistic):               0.00
Time:                        16:39:14   Log-Likelihood:                -14080.
No. Observations:               12003   AIC:                         2.818e+04
Df Residuals:                   11994   BIC:                         2.824e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
X1             0.0674      0.011      6.146      0.0

•모델의 설명력(R-squared)
◾즉, 결정계수가 0.967이라면 Y의 변동 중 96.7%가 회귀모형에 의해 설명되고 나머지 3.3%는 모형에 포함되지 않는 다른 요인들로 설명됨.
◾단순 선형회귀에서는 상관계수r의 제곱과 정확히 일치함. r^2 = R^2
◾r=0.983 -> r^2=0.967 -> R^2 = 0.967 X와 Y의 상관관계가 높을 수록 모형에 의해 데이터가 많이 설명될 수 있다.

In [23]:
from sklearn import preprocessing
import pandas as pd

X=pd.DataFrame(data, columns=["X1","X2","X3","X5","X6","X7","X8","X9","X10"])
Y=pd.DataFrame(data, columns=["Y2"])

In [24]:
# 만족도와 추천의사 회귀분석
linear_regression = sm.OLS(Y, X) # (종속변수, 독립변수)

# OLS 회귀모형 피팅
result_data = linear_regression.fit()

# 회귀분석 결과 요약
print(result_data.summary())

                            OLS Regression Results                            
Dep. Variable:                     Y2   R-squared:                       0.979
Model:                            OLS   Adj. R-squared:                  0.979
Method:                 Least Squares   F-statistic:                 6.313e+04
Date:                Sat, 09 Jun 2018   Prob (F-statistic):               0.00
Time:                        16:39:41   Log-Likelihood:                -11415.
No. Observations:               12003   AIC:                         2.285e+04
Df Residuals:                   11994   BIC:                         2.291e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
X1             0.0724      0.009      8.238      0.0

- 모델의 설명력(R-squared)
    - 즉, 결정계수가 0.979이라면 Y의 변동 중 97.9%가 회귀모형에 의해 설명되고 나머지 2.1%는 모형에 포함되지 않9는 다른 요인들로 설명됨.
    - 단순 선형회귀에서는 상관계수r의 제곱과 정확히 일치함. r^2 = R^2
    - r=0.989 -> r^2=0.979 -> R^2 = 0.979 X와 Y의 상관관계가 높을 수록 모형에 의해 데이터가 많이 설명될 수 있다.